src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -82,9 +83,19 @@ public PdbSourceDocumentMetadataAsSourceFileProvider(IPdbFileLocatorService pdbF
                     return null;
             }
 
+            Encoding? defaultEncoding = null;
+            if (pdbCompilationOptions.TryGetValue("default-encoding", out var encodingString))
+            {
+                defaultEncoding = Encoding.GetEncoding(encodingString);
```

Comment:
>GetEncoding [](http://example.com/codeflow?start=43&length=11)

May throw ArgumentException

Reply:
Since we own all the involved bits it should be fine to let an exception be exceptional here. We put the wrong information in the pdb